include/range/v3/utility/concepts.hpp

```diff
@@ -655,8 +626,13 @@ namespace ranges
         template<typename T, typename...Us>
         using Constructible = concepts::models<concepts::Constructible, T, Us...>;
 
-        template<typename T>
-        using DefaultConstructible = concepts::models<concepts::DefaultConstructible, T>;
+        template<typename T, typename...Us>
+        using ConstructibleRef =
+            meta::apply<
+                meta::if_<std::is_reference<T>,
+                    meta::quote<std::is_constructible>,
+                    meta::quote<Constructible>>,
+                T, Us...>;
```

Comment:
Everything in the hierarchy between Destructible - Regular is specific to object types. Constructible will reject reference types because of (a) the `new` requirement, (b) Destructible's requirement for `t.~T()`, and (c) Destructible's formation of a pointer-to-T.

ConstructibleRef - which I probably should have named `ConstructibleObjectOrReference`- exists to constrain variables of type T to have either an object type that is Constructible from Args or a reference type that can be bound to Args. It's used by the constructors of pair and tuple to ensure both that the members can be constructed from the constructor arguments and that the constructed pair/tuple will be Destructible. If we simply replace ConstructibleRef uses with is_constructible the guarantee of a Destructible tuple/pair is lost, violating my "Construction implies destruction" design axiom.